lib/guardian/post_guardian.rb

```diff
@@ -280,6 +283,11 @@ def can_see_deleted_post?(post)
     post.deleted_by_id == @user.id && @user.has_trust_level?(TrustLevel[4])
   end
 
+  def can_see_hidden_post?(post)
+    return false if @user.anonymous?
```

Comment:
You can just do `if anonymous?`, see https://github.com/discourse/discourse/blob/24ec06ff85c7acbad9621092b5e50eec2ede7b83/lib/guardian.rb#L96-L98